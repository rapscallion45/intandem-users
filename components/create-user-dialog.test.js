import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { createStore, applyMiddleware } from 'redux';
import thunkMiddleware from 'redux-thunk';
import rootReducer from '../redux/reducers/reducers';
import CreateUserDialog from './create-user-dialog';

const middleware = [thunkMiddleware];

describe('Create User Dialog', () => {
  it('Renders correctly enzyme', () => {
    /* Arrange */
    const testStore = createStore(rootReducer, {}, applyMiddleware(...middleware));
    const dialogOpen = true;

    /* Act */
    const wrapper = shallow(
      <Provider store={testStore}>
        <CreateUserDialog open={dialogOpen} />
      </Provider>
    );

    /* Assert */
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  describe('Dialog Text', () => {
    it('Should render title text', () => {
      /* Arrange */
      const testStore = createStore(rootReducer, {}, applyMiddleware(...middleware));
      const titleText = 'Create New User';
      const dialogOpen = true;

      /* Act */
      const wrapper = render(
        <Provider store={testStore}>
          <CreateUserDialog open={dialogOpen} />
        </Provider>
      );

      /* Assert */
      expect(wrapper.getByText(titleText)).toBeInTheDocument();
    });
  });

  describe('Dialog Operation', () => {
    it('Should not render if dialog open prop not set', () => {
      /* Arrange */
      const testStore = createStore(rootReducer, {}, applyMiddleware(...middleware));
      const titleText = 'Create New User';
      const dialogOpen = false;

      /* Act */
      const wrapper = render(
        <Provider store={testStore}>
          <CreateUserDialog open={dialogOpen} />
        </Provider>
      );

      /* Assert */
      expect(wrapper.queryByText(titleText)).toBeNull();
    });
  });
});
