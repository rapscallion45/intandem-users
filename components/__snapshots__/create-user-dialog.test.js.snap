// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create User Dialog Renders correctly enzyme 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "@@observable": [Function],
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Object {
        "addNestedSub": [Function],
        "getListeners": [Function],
        "handleChangeWrapper": [Function],
        "isSubscribed": [Function],
        "notifyNestedSubs": [Function],
        "onStateChange": [Function],
        "trySubscribe": [Function],
        "tryUnsubscribe": [Function],
      },
    }
  }
>
  <CreateUserDialog
    open={true}
  />
</ContextProvider>
`;
